packages/near-membrane-dom/src/browser-realm.ts

```diff
@@ -150,6 +150,22 @@ function createIframeVirtualEnvironment(
     // We intentionally skip remapping Window.prototype because there is nothing
     // in it that needs to be remapped.
     env.lazyRemapProperties(blueRefs.EventTargetProto, blueRefs.EventTargetProtoOwnKeys);
+    // Install default TrustedTypes policy in the virtual environment.
+    // This is required for eval and Function constructor to run.
+    // We automatically sign any input received as arguments
+    // since they run in a secure environment.
+    //
+    // We do not define hooks for createScriptURL and createHTML.
+    // The VirtualEnvironment doesn't have built in means to protect these
+    // DOM APIs from XSS attacks. Use distortions to intercept, sanitize
+    // and sign content for script URLs and HTML strings.
+    // @ts-ignore trustedTypes does not exist on GlobalObject
+    if (typeof redWindow.trustedTypes !== 'undefined') {
+        // @ts-ignore trustedTypes does not exist on GlobalObject
+        redWindow.trustedTypes.createPolicy('default', {
```

Comment:
this should be done prior to remap rather than after remap, in which case you don't have to use the intrinsics trick.

Reply:
The remap will just overwrite everything, I did that and I couldn't get the policy to be installed. Adding it as an intrinsic prevents remapping which is a problem when we define our default policy. So this is the only place to do it and it has to be an intrinsic even though the language doesn't define it as one it was the only way I found to prevent it from being remapped and interfering with the rest of the things (remapping & main window default policy).

Reply:
the policy is installed before the remap, that is the key here, after that, who cares much about what they see. It is true that you might also need to distort the defaultPolicy to do nothing since the default will kick-in if you return a string, it is sort of redundant.